sdk/spring/azure-spring-cloud-stream-binder-eventhubs/src/main/java/com/azure/spring/eventhub/stream/binder/config/EventHubBinderHealthIndicatorConfiguration.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.eventhub.stream.binder.config;
+
+import com.azure.spring.eventhub.stream.binder.EventHubHealthIndicator;
+import com.azure.spring.eventhub.stream.binder.EventHubMessageChannelBinder;
+import com.azure.spring.integration.eventhub.api.EventHubClientFactory;
+import org.springframework.boot.actuate.autoconfigure.health.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnClass(name = "org.springframework.boot.actuate.health.HealthIndicator")
+@ConditionalOnEnabledHealthIndicator("binders")
+class EventHubBinderHealthIndicatorConfiguration {
+
+    @Bean
+    EventHubHealthIndicator eventHubHealthIndicator(EventHubMessageChannelBinder binder,
```

Comment:
I think we need add a conditional on missing bean annotation here, so this can be overrided?

Reply:
The binders could take multiple health contributors as args: 

https://github.com/spring-cloud/spring-cloud-stream/blob/93acdcf85ad3d47936d71cd5bcc38eefb0d4290a/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindersHealthIndicatorAutoConfiguration.java#L81-L82

https://github.com/spring-cloud/spring-cloud-stream/blob/93acdcf85ad3d47936d71cd5bcc38eefb0d4290a/spring-cloud-
stream/src/main/java/org/springframework/cloud/stream/config/BindersHealthIndicatorAutoConfiguration.java#L97

But I'm not sure when is it possible for users to override this health indicator in a binder's context?